### PR TITLE
fix(material-experimental/mdc-slider): update layout when container resizes

### LINKED ISSUE #24590: bug(mat-mdc-slider): thumb should use % instead of px
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

_No response_

### Description

https://stackblitz.com/edit/angular-us2jqe-a7chp1?file=src%2Fapp%2Fslider-overview-example.ts

The thumb is not correctly positioned when resizing the container of a mat-mdc-slider. However, the track is.

### Reproduction

Steps to reproduce:
1. Have a mat-mdc-slider with CSS `width: 100%`
2. Resize it's container with any means other than window resize (i.e., a UX could have a panel that pushes content meaning the container will be less wide)

Screen Shot 2022-03-14 at 4 21 19 PM

### Expected Behavior

Thumb and track are always aligned

### Actual Behavior

Thumb and track stop being aligned

### Environment

- Angular: 13
- CDK/Material: 13
- Browser(s): All
- Operating System (e.g. Windows, macOS, Ubuntu): All

### PR DESCRIPTION
**Pull request by crisbeto**
Currently the layout of the slider can look broken, because the thumb positions don't update when the size of the container changes.

These changes add some extra logic to trigger the resize in such cases.

Fixes #24590.

### CODE REVIEW

File: src/material-experimental/mdc-slider/slider.ts [hunk 1]
```diff
@@ -919,6 +929,31 @@ export class MatSlider
...
+    if (typeof ResizeObserver === 'undefined' || !ResizeObserver) {
+      return;
+    }
+
+    // MDC only updates the slider when the window is resized which
+    // doesn't capture changes of the container itself. We use a resize
+    // observer to ensure that the layout is correct (see #24590).
+    this._ngZone.runOutsideAngular(() => {
+      // The callback will fire as soon as an element is observed and
+      // we only want to know after the initial layout.
```
Comment (ajafff) [inline src/material-experimental/mdc-slider/slider.ts:944]: this is actually what `ResizeObserver` already does. The callback is not executed synchronously, but after initial layout is done, see https://medium.com/@paul_irish/requestanimationframe-scheduling-for-nerds-9c57f7438ef4

if you actually want to ignore the first layout (i.e. the first callback invocation), the comment is misleading

Comment (crisbeto) [reply src/material-experimental/mdc-slider/slider.ts:944]: I was playing around with resizing the browser and it definitely felt like it was emitting for each pixel.

Comment (devversion) [reply src/material-experimental/mdc-slider/slider.ts:944]: You could potentially link to https://www.w3.org/TR/resize-observer/#:~:text=Observation%20will%20fire-,when,-observation%20starts%20if

**Review by devversion — APPROVED** [review]
LGTM

**Pull request merged**